plugins/GSdx/GSDevice11.cpp

```diff
@@ -432,9 +433,18 @@ void GSDevice11::SetExclusive(bool isExcl)
 	}
 }
 
+void GSDevice11::SetVSync(int vsync)
+{
+	if (m_vsync == vsync)
+		return;
+
+	__super::SetVSync(vsync);
+	m_vsyncinterval = vsync ? 1 : 0;
```

Comment:
Maybe just `GSDevice::SetVSync(std::abs(vsync));` or even `m_vsync = std::abs(vsync);`?

Also, your commit message is a bit inaccurate since the Present function only accepts 0-4 (I did a quick check).

Reply:
> Also, your commit message is a bit inaccurate since the Present function only accepts 0-4 (I did a quick check).

You mean this part ``"the present function
of swapchain only accepts unsigned integers as parameter"``? I just wrote what's given in [Microsoft documentation](https://msdn.microsoft.com/en-us/library/windows/desktop/bb174576(v=vs.85).aspx).

``` cpp
HRESULT Present(
   UINT SyncInterval,
   UINT Flags
);
```
Also I said ``"only accepts unsigned integers as parameter"`` which implies that values exclusive to signed integers are not accepted. Doesn't necessarily mean that all the unsigned integers are accepted as parameters.

> Maybe just GSDevice::SetVSync(std::abs(vsync)); or even m_vsync = std::abs(vsync);?

``std::abs`` isn't a good idea, if Gregory adds a new -2 mode to the VSync combobox then we'll pass 2 as the Vsync interval for swapchain present function and it won't be easy to find this altered behavior as it's just a delay in presentation instead of something as noticeable as #2050.

I guess what you're actually against is the addition of ``m_vsyncinterval``, I think it's useful for certain reasons but I can certainly remove it if you want it to be removed, don't really feel like arguing about it right now.

Reply:
0-4 is more specific and is also in the documentation.

I doubt another vsync mode will be added, but I guess you're right on `std::abs`. And yes, I don't get why `m_vsyncinterval` was used instead of `m_vsync`.

Reply:
> if Gregory adds a new -2 mode

Mhh.. Why should he? Adaptive is supposed to be -1.

Reply:
-2 is legal in GL as 2. Besides, it could also be interesting to keep some special value for strange mode (such as dwm).

Reply:
.-.
Why should we always make this assumption for the user?